### ISSUE TITLE
what about two factor authentication like google authenticator for clients

### ISSUE THREAD

**Issue by renepickhardt** [shesek/spark-wallet#5]
I imagine consumers loosing there cellphones, leaving their computer / wallet unprotected... at least for making payments there should be something like this (:

**Comment by shesek** [comment]
I think that a good first step could be requiring a PIN number for sending payments. With cordova on mobile, spark could also integrate with the device biometric authentication (there are plugins that makes that pretty easy).

2FA with Google Authenticator for every payment seems a bit much, to the point of seriously degrading UX. But I think that maybe having some daily spending limit, where below that you just need the PIN and above it you require an additional confirmation (2FA, email, SMS, a digital signature from some HW device, etc), could make a lot of sense.

But given that users should not have significant amounts in lightning right now (especially not with Spark, but even with the underlying lightning implementations directly), I think that a daily spending limit is probably not very high priority and that a simpler PIN code could be sufficient for now (and relatively straightforward to implement).

**Comment by d4amenace** [comment]
I agree with shesek, a pin code for sending is not only a good idea but needed.  This would enable merchants to allow remote employees to accept payments easily and give owners peace of mind that employees aren't stealing. 

On that same note, shesek, i think the ability to hide the value in the channel is needed so everyone in an organization won't know the full funded amount of the node.  Or perhaps, adding a pin code for the amount would be an option?

**Comment by shesek** [comment]
> @shesek Should I open a seperate issue for this?

Yes, please. I think that for the use-case you described, a receive-only mode would be more useful than a PIN.
